pkg/referrer/referrer.go

```diff
@@ -93,7 +98,7 @@ func (r *referrer) checkReferrer(ctx context.Context, ref string, manifestDigest
 	}
 
 	desc, err := handle()
-	if err != nil && r.remote.RetryWithPlainHTTP(ref, err) {
+	if err != nil && retry && r.remote.RetryWithPlainHTTP(ref, err) {
```

Comment:
the `RetryWithPlainHTTP` func will avoid retry if the error is not network related.